.gitlab-ci.yml

```diff
@@ -0,0 +1,34 @@
+image: 'rust:latest'
+
+stages:
+  - fmt
+  - test
+
+variables:
+  CARGO_HOME: $CI_PROJECT_DIR/cargo
+  APT_CACHE_DIR: $CI_PROJECT_DIR/apt
+  RUSTFLAGS: "-D warnings"
+
+before_script:
+  - apt-get update -yq
+  - rustup component add rustfmt-preview
+  - apt-get install -o dir::cache::archives="$APT_CACHE_DIR" -y alsa-utils libasound2-dev cmake git libasound2-dev libx11-xcb-dev libssl-dev cmake libfreetype6-dev libexpat1-dev libxcb1-dev python3 build-essential libsdl2-dev
+
+check formatting:
+  stage: fmt
+  script:
+    - cargo fmt --version
+    - cargo fmt -- --check
+
+run tests:
+  stage: test
+  script:
+    - rustc --version
+    - cargo --version
+    - cargo test --verbose --features sdl_controller,profiler
```

Comment:
I think we need to test json and profiler, too.

Reply:
profiler is in there, I think. or is it a different profiler?

Reply:
Um, yeah, missed it.

Reply:
I think it could just be `--all-features`.

Reply:
Changed.